src/java.base/share/classes/com/sun/crypto/provider/AES_Crypt.java

```diff
@@ -0,0 +1,1418 @@
+            0x8D13C2F6, 0x841EC9F8, 0xBB3DF8D2, 0xB230F3DC, 0xA927EECE,
+            0xA02AE5C0, 0x47B13C7A, 0x4EBC3774, 0x55AB2A66, 0x5CA62168,
+            0x63851042, 0x6A881B4C, 0x719F065E, 0x78920D50, 0x0FD9640A,
+            0x06D46F04, 0x1DC37216, 0x14CE7918, 0x2BED4832, 0x22E0433C,
+            0x39F75E2E, 0x30FA5520, 0x9AB701EC, 0x93BA0AE2, 0x88AD17F0,
+            0x81A01CFE, 0xBE832DD4, 0xB78E26DA, 0xAC993BC8, 0xA59430C6,
+            0xD2DF599C, 0xDBD25292, 0xC0C54F80, 0xC9C8448E, 0xF6EB75A4,
+            0xFFE67EAA, 0xE4F163B8, 0xEDFC68B6, 0x0A67B10C, 0x036ABA02,
+            0x187DA710, 0x1170AC1E, 0x2E539D34, 0x275E963A, 0x3C498B28,
+            0x35448026, 0x420FE97C, 0x4B02E272, 0x5015FF60, 0x5918F46E,
+            0x663BC544, 0x6F36CE4A, 0x7421D358, 0x7D2CD856, 0xA10C7A37,
+            0xA8017139, 0xB3166C2B, 0xBA1B6725, 0x8538560F, 0x8C355D01,
+            0x97224013, 0x9E2F4B1D, 0xE9642247, 0xE0692949, 0xFB7E345B,
+            0xF2733F55, 0xCD500E7F, 0xC45D0571, 0xDF4A1863, 0xD647136D,
+            0x31DCCAD7, 0x38D1C1D9, 0x23C6DCCB, 0x2ACBD7C5, 0x15E8E6EF,
+            0x1CE5EDE1, 0x07F2F0F3, 0x0EFFFBFD, 0x79B492A7, 0x70B999A9,
+            0x6BAE84BB, 0x62A38FB5, 0x5D80BE9F, 0x548DB591, 0x4F9AA883,
+            0x4697A38D,
+    };
+
+    /**
+     * Return the block cipher size.
+     *
+     * @return the AES block cipher size (in bytes).
+     */
+    int getBlockSize() {
+        return AESConstants.AES_BLOCK_SIZE;
+    }
+
+    /**
+     * Verifies if the length argument is a valid key size or not.
+     *
+     * @param len the size of the key (in bytes) to be validated.
+     *
+     * @return {@code true} if the size of the key is valid else {@code false}.
+     */
+    static boolean isKeySizeValid(int len) {
+        for (int kLength : AESConstants.AES_KEYSIZES) {
+            if (len == kLength) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check algorithm and initalize round keys for decryption and encryption.
+     *
+     * @param decrypting [in] indicates if encrypting ({@code false}) or
+     *      decrypting ({@code true}).
+     * @param algorithm [in] the case insentive string name for the AES cipher.
+     * @param key [in] the symmetric key byte array for encryption/decryption.
+     *
+     * @throws InvalidKeyException if an incorrect algorithm is given or if
+     * an invalid key size is provided.
+     */
+    void init(boolean decrypting, String algorithm, byte[] key)
+            throws InvalidKeyException {
+        int nk;
+        int decrypt = decrypting ? 1 : 0;
+
+        if (!algorithm.equalsIgnoreCase("AES")
+                && !algorithm.equalsIgnoreCase("Rijndael")) {
+            throw new InvalidKeyException ("Invalid algorithm name.");
+        }
+        if (key.length == AESConstants.AES_KEYSIZES[0]) {
+            rounds = AES_128_ROUNDS;
+            nk = AESConstants.AES_KEYSIZES[0]/WB;
```

Comment:
Looks like we can get rid of `nk` as the `genRKeys(byte[])` method can calculate/derive it based on the specified `key` argument, i.e. `key.length >> 2` or `key.length / WB`